src/bindings/html/observer.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+import { translateFragment } from './dom';
+
+const observerConfig = {
+  attributes: true,
+  characterData: false,
+  childList: true,
+  subtree: true,
+  attributeFilter: ['data-l10n-id', 'data-l10n-args']
+};
+
+const observers = new WeakMap();
+
+export function initMutationObserver(view, onMutations) {
+  observers.set(view, {
+    roots: new Set(),
+    observer: new MutationObserver(
+      mutations => onMutations(view, mutations)),
+  });
+}
+
+export function translateRoots(view, langs) {
+  return Promise.all(
+    [...observers.get(view).roots].map(
+      root => translateFragment(view, langs, root)));
+}
+
+export function disconnect(view) {
+  const obs = observers.get(view);
+  if (obs) {
+    obs.observer.disconnect();
+  }
+}
+
+export function reconnect(view) {
+  const obs = observers.get(view);
+  if (obs) {
+    obs.roots.forEach(
+      root => obs.observer.observe(root, observerConfig));
+  }
+}
+
+export function observeRoot(view, root) {
+  const obs = observers.get(view);
+  obs.roots.add(root);
+  obs.observer.observe(root, observerConfig);
+}
+
+export function disconnectRoot(view, root) {
```

Comment:
can you merge `disconnect` and `disconnectRoot`? Also, you probably want the guard on `observers.get` result anyway.